ezidapp/management/commands/proc-link-checker-update.py

```diff
@@ -96,11 +97,21 @@ def run(self):
                                 si2 = ezidapp.models.identifier.SearchIdentifier.objects.get(
                                     identifier=si.identifier
                                 )
+                                # get before values for OpenSearchDoc for link/issues
+                                before_link_is_broken = si2.linkIsBroken
+                                before_has_issues = si2.hasIssues
+
                                 si2.linkIsBroken = newValue
                                 si2.computeHasIssues()
-                                si2.save(update_fields=["linkIsBroken", "hasIssues"])
-                                open_s = OpenSearchDoc(identifier=si2)
-                                open_s.update_link_issues(link_is_broken=si2.linkIsBroken, has_issues=si2.hasIssues)
+
+                                # only update if the record is "dirty" and the values have changed
+                                if before_link_is_broken != si2.linkIsBroken or before_has_issues != si2.hasIssues:
```

Comment:
This might have been covered  by line line 91 "if newValue is not None:" .

Reply:
You might be right, though I don't understand if this is really checking if everything changed.  It looks like newValue = none and then  link_checker.identifier > search.identifer sets it to false and otherwise sets it to True ?

This code isn't very clear, though it looks like linkIsBroken gets set to newValue, so it seems it may get changed at line 104.  I assume si2.computeHasissues() also possibly changes the value of "hasIssues" (line 105).

I'm still not really sure it only updates when values have changed.  Probably not a huge deal, but trying to avoid unnecessary updates to both the database and OpenSearch if the values are not different than they already were.

Reply:
I removed the conditional update only when values are changed.